### PR TITLE
Feature/5 convert to typescript

### LINKED ISSUE #5: Convert to typescript
let's get all the help we can :)

### PR DESCRIPTION
**Pull request by wooneh**
Closes #5 

### List of Changes
* Converted JS to TS
* Added TS types and fixed errors
* Added linting and code formatting
* Added git hooks ([Husky](https://typicode.github.io/husky/#/))

To enable git hooks, run `yarn prepare`. This enables the `precommit` and `prepush` hooks, which will run the linter and tests respectively before running the git commands.

For autoformatting, I use:
```
{
    "editor.codeActionsOnSave": {
        "source.fixAll.eslint": true
    }
}
```

If using VS code, add that to `.vscode/settings.json` and install [ESLint ](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)and [Prettier ](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) extensions.

This will autofix any formatting issues and also highlight errors.

### CODE REVIEW

File: .husky/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_
```
Comment (chrisbc) [inline .husky/.gitignore:1]: @wooneh is this file going to be used for something - or can we drop it?

File: relay.config.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 module.exports = {
   // ...
   // Configuration options accepted by the `relay-compiler` command-line tool and `babel-plugin-relay`.
-  src: "./src",
-  schema: "./data/schema.graphql",
-  exclude: ["**/node_modules/**", "**/__mocks__/**", "**/__generated__/**"],
-}
+  src: './src',
```
Comment (chrisbc) [inline relay.config.js:5]: the linter at work?

File: src/custom.d.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+declare module 'babel-plugin-relay/macro' {
```
Comment (chrisbc) [inline src/custom.d.ts:1]: what is this module for?

**Review by chrisbc — COMMENTED** [review]
@wooneh - looks pretty good. I can see the typing on props being a big help :)

Before I approve I'm going to check out your branch and test the app manually, and try our the vscode and extensions since this is such an influential PR :)

File: .husky/.gitignore (hunk 1 above)
Comment (wooneh) [reply .husky/.gitignore:1]: This is autogenerated by husky, so we should keep it.

File: relay.config.js (hunk 2 above)
Comment (wooneh) [reply relay.config.js:5]: Yes, I could set it to ignore config files however I've allowed it here

File: src/custom.d.ts (hunk 3 above)
Comment (wooneh) [reply src/custom.d.ts:1]: This is a file that exports types that Typescript has trouble detecting (usually due to the module itself and not Typescript). [Related issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/35707)

**Pull request merged**